docs/notebook_version_standardizer.py

```diff
@@ -0,0 +1,77 @@
+import click
+import json
+import os
+
+DOCS_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), "source")
+
+
+def _get_python_version(notebook):
+    with open(notebook, "r") as f:
+        source = json.load(f)
+        version = source['metadata']['language_info']['version']
+    return version
+
+
+def _standardize_python_version(notebook, desired_version='3.7.4'):
+    with open(notebook, "r") as f:
+        source = json.load(f)
+        source['metadata']['language_info']['version'] = desired_version
+    json.dump(source, open(notebook, "w"), indent=1)
+
+
+def _get_ipython_notebooks(docs_source):
+    directories_to_skip = ["_templates", "generated", ".ipynb_checkpoints"]
+    notebooks = []
+    for root, _, filenames in os.walk(docs_source):
+        if any(dir_ in root for dir_ in directories_to_skip):
+            continue
+        for filename in filenames:
+            if filename.endswith('.ipynb'):
+                notebooks.append(os.path.join(root, filename))
+    return notebooks
+
+
+def _get_notebooks_with_different_versions(notebooks, desired_version='3.7.4'):
+    different_versions = []
+    for notebook in notebooks:
+        version = _get_python_version(notebook)
+        if version != desired_version:
+            different_versions.append(notebook)
+    return different_versions
+
+
+def _standardize_versions(notebooks, desired_version='3.7.4'):
+    for notebook in notebooks:
+        _standardize_python_version(notebook, desired_version)
+
+
+@click.group()
+def cli():
+    """no-op"""
+
+
+@cli.command()
+@click.option('--desired-version', default='3.7.4', help='python version that all notebooks should match')
```

Comment:
Perhaps we should standardize to the latest 3.8? I guess it doesn't really matter lol, unless there's some ugly version-dependent docs bug which pops up haha. I remember we surveyed the group and most people were running different python versions locally so that doesn't help us decide here 😆

Reply:
I'm going for 3.8.2 since that was the version of 3.8 that was in the notebooks before this PR. The good thing is that we can pass in the desired version in the makefile in the future